lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -919,11 +912,38 @@ namespace Js
         Var trapResult = getEnumeratorMethod->CallFunction(arguments);
         threadContext->SetImplicitCallFlags((Js::ImplicitCallFlags)(saveImplicitCallFlags | ImplicitCall_Accessor));
 
-        if (!JavascriptOperators::IsObject(trapResult))
+        if (!JavascriptOperators::IsArray(trapResult))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("enumerate"));
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("ownKeys"));
         }
-        *enumerator = IteratorObjectEnumerator::Create(scriptContext, trapResult);
+
+        JavascriptArray* arrTrapResult = JavascriptArray::FromVar(trapResult);        
+        uint32 len = arrTrapResult->GetLength();
```

Comment:
hmm interesting. JavascriptOperators::IsArray() is true for proxy over array. you can't cast it to JavascriptArray if the trapResult is a proxy.

---
In reply to: [63961681](https://github.com/Microsoft/ChakraCore/pull/976#discussion_r63961681) [](ancestors = 63961681)